gu-client/src/async.rs

```diff
@@ -40,17 +42,17 @@ pub struct HubConnection {
 
 #[derive(Debug)]
 struct HubConnectionInner {
-    url: String,
+    url: Url,
 }
 
 impl HubConnection {
     /// creates a hub connection from a given address:port, e.g. 127.0.0.1:61621
-    pub fn from_addr<T: Into<String>>(addr: T) -> HubConnection {
-        HubConnection {
-            hub_connection_inner: Arc::new(HubConnectionInner {
-                url: format!("http://{}/", addr.into()),
-            }),
-        }
+    pub fn from_addr<T: Into<String>>(addr: T) -> Result<HubConnection, Error> {
+        Url::parse(&format!("http://{}/", addr.into()))
+            .map_err(|_| Error::InvalidAddress)
```

Comment:
Why are we discarding the original error causes? It may make the issues hard to solve for the API users

Reply:
(Messages are stored as strings like in the gu-envman-api crate so that there are no external dependencies; it can be changed in the future.)

Reply:
Ok. Replaced strings with original errors.